src/server/scripts/EasternKingdoms/Stratholme/boss_baroness_anastari.cpp

```diff
@@ -52,18 +53,18 @@ struct boss_baroness_anastari : public BossAI
 
     void Reset() override
     {
-        if (!me->IsVisible())
+        if (me->HasAura(SPELL_POSSESS_INV))
         {
-            if (Player* possessedTarget = ObjectAccessor::FindConnectedPlayer(_possessedTargetGuid))
+            if (Player* possessedTarget = ObjectAccessor::GetPlayer(me->GetMap(), _possessedTargetGuid))
             {
                 if (possessedTarget->IsAlive())
                 {
                     possessedTarget->RemoveAurasDueToSpell(SPELL_POSSESSED);
```

Comment:
iirc instancemap has some method for removing auras by id from all players, could be used here
also don't forget to reset guid

Reply:
Won't be overwritten by ` if (Unit* possessTarget = SelectTarget(SELECT_TARGET_RANDOM, 1, 0, true, false))` ?

Reply:
Yeah, I guess so. Still cleaner if you don't leave anything "lying around" from the previous attempt.

Reply:
I would just add `_possessedTargetGuid.Clear();` after https://github.com/TrinityCore/TrinityCore/pull/21687/files#diff-db60b62d3ff54ba02d4f8c161fe09c8dR133 and I guess in `Reset()` also?